shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -358,6 +373,9 @@ - (void)shutDownEngine {
     NSLog(@"Could not de-initialize the Flutter engine: error %d", result);
   }
 
+  // Balancing release for the retain in the task runner dispatch table.
+  CFRelease((CFTypeRef)self);
```

Comment:
The declaration comment above can't possibly be correct with this change, because dealloc won't be called as long as this reference exists. So it seems like this makes it impossible to clean up engine instances.